File: frontend/src/api/position.js
import http from './http'

export const getPositions = (params) => http.get('/positions', { params })
export const getPosition = (id) => http.get(`/positions/${id}`)
export const createPosition = (data) => http.post('/positions', data)
export const updatePosition = (id, data) => http.put(`/positions/${id}`, data)
export const deletePosition = (id) => http.delete(`/positions/${id}`)
export const getPositionMenuPermissions = (id) => http.get(`/positions/${id}/menu-permissions`)
export const setPositionMenuPermissions = (id, data) => http.put(`/positions/${id}/menu-permissions`, data)

File: backend/routes/position.go
package routes

import (
	"oa-system/handlers"

	"github.com/gin-gonic/gin"
)

func positionRoutes(rg *gin.RouterGroup) {
	rg.GET("/positions", handlers.GetPositions)
	rg.GET("/positions/:id", handlers.GetPosition)
	rg.POST("/positions", handlers.CreatePosition)
	rg.PUT("/positions/:id", handlers.UpdatePosition)
	rg.DELETE("/positions/:id", handlers.DeletePosition)
	rg.PUT("/positions/:id/employees", handlers.SetPositionEmployees)
	rg.GET("/positions/:id/menu-permissions", handlers.GetPositionMenuPermissions)
	rg.PUT("/positions/:id/menu-permissions", handlers.SetPositionMenuPermissions)
}
